drivers/hmis/app/models/hmis/filter/enrollment_filter.rb

```diff
@@ -13,6 +13,7 @@ def filter_scope(scope)
       yield_self(&method(:with_bed_night_on_date)).
       yield_self(&method(:with_project_types)).
       yield_self(&method(:with_search_term)).
+      yield_self(&method(:with_household_tasks)).
       yield_self(&method(:clean_scope))
```

Comment:
I am actually not able to reproduce that if I put in 3/29/2022 for an entry date, in the app or in tests. I switched to using `1 month` as the interval in hopes that that might help. However, if I had to guess I think it'd be from the `make_date(...)` piece, as it would be possible to turn an initial date of 2/29/2020 into 2/29/2023 there. Still, setting a date of 2/29/2020 for entry date doesn't crash anything for me. Maybe we can look at it together?

Reply:
Okay, so I think I might have found a solution. Instead of creating the date directly, I've made it create a date for the first of the month, then add an interval of days to get it to the same day of the month. This might mean that a leap year date of 2/29 will be "due" a day before or after the actual 30 day window on later non-leap years, but that seems like it's maybe okay? I also switched back to `30 days` instead of `1 month` for the interval. Take a look again and see if you still have this problem when you pull down the latest code.

Reply:
Okay I'll try it out! Thanks!  Its totally fine if leap year dates are off by one.

Reply:
Alright I'm not getting errors anymore, thought I'm not totally sure I understand why that is. I tried out several configurations including acting like this year is a leap year (2024), and it seemed to work

Reply:
So I think the issue was that `make_date` was creating an invalid date. So 2/29/2020 is a valid date, but if we use `make_date` to shift the year, then we end up with 2/29/2023, which is not valid. I don't know why `make_date` wasn't erroring out for me in these cases, but it's the only thing I could imagine was creating invalid dates for you and causing the errors.

So to deal with that, we just use `make_date` to create a date on the first day of the anniversary month (e.g. `2/1/2023`), then extract the day of month from the entry date and add it back to the date minus one day. That should result in the same month and day for any other date (e.g. 4/22/2020 -> 4/1/2023 + 21 days == 4/22/2023). For 2/29 it should result in 2/29 on leap years, and 3/1 on non-leap years.

This avoids the possibility of creating an invalid 2/29 date on on a non-leap year, assuming postgres's date math is correct, which I have to assume it generally should be.